dali/kernels/common/block_setup.h

```diff
@@ -88,7 +104,10 @@ class BlockSetup {
     return shape;
   }
 
-
+  /**
+   * @brief Check if `out_shape` matches the number of channels of `in_shape` and spatial dimensions
+   * of `output_sizes` as if it was generated with GetOutputShape(in_shape, output_sizes).
+   */
```

Comment:
I'm wondering, if this comment is not simple description of the implementation detail. Shouldn't it be rather something like:
> Check, if this BlockSetup may produce an output shape, given an input shape 

?

Reply:
I think it's a definition that makes you look for what happens in other function.